### PR TITLE
Click CLI profiles directory

### PR DESCRIPTION
**Pull request by dbeatty10**
### Description

Bring the Click CLI back into parity with the current CLI per https://github.com/dbt-labs/dbt-core/pull/5717#discussion_r975717672.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] ~This PR includes tests, or tests are not required/relevant for this PR~ manually verified behavior
- [x] ~I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or~ docs changes are not required/relevant for this PR
- [x] ~I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)~

### CODE REVIEW

File: core/dbt/cli/resolvers.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        # Use the directory if there is a profiles.yml file
+        if (cwd / "dbt_project.yml").exists():
+            return Path(cwd)
+        cwd = cwd.parent
+
+    return Path.cwd()
+
+
+def default_profiles_dir():
+    default_profiles_dir = Path.home() / ".dbt"
```
Comment (iknox-fa) [inline core/dbt/cli/resolvers.py:18]: This looks great, but we can tighten it up a bit by just using a conditional return here.
```
def default_profiles_dir():
    return Path.cwd() if (Path.cwd() / "profiles.yml").exists() else Path.home() / ".dbt"
````

File: core/dbt/cli/resolvers.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from pathlib import Path
+
+
+def default_project_dir():
```
Comment (iknox-fa) [inline core/dbt/cli/resolvers.py:4]: Again, this looks great but we can take advantage of some fancy pathlib stuff to make things easier on ourselves.

```
def default_project_dir():
    paths = list(Path.cwd().parents)
    paths.insert(0, Path.cwd()) # adds cwd to the list of paths
    return next((x for x in paths if (x / "dbt_project.yml").exists()), Path.cwd()) 
```

**Review by iknox-fa — APPROVED** [review]
@dbeatty10 
I made a few comments, but they're mostly FYI-- this code works and is good to merge IMO

File: core/dbt/cli/resolvers.py (hunk 2 above)
Comment (dbeatty10) [reply core/dbt/cli/resolvers.py:4]: @iknox-fa What is your reaction to this tweak? (Untested)
```
def default_project_dir():
    default_dir = Path.cwd()
    paths = list(default_dir).extend(default_dir.parents)
    return next((x for x in paths if (x / "dbt_project.yml").exists()), default_dir)
```

**Pull request merged**